controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -288,60 +290,168 @@ func (r *KubeadmControlPlaneReconciler) generateKubeadmConfig(ctx context.Contex
+			Kind:       "Machine",
+		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        names.SimpleNameGenerator.GenerateName(kcp.Name + "-"),
-			Namespace:   kcp.Namespace,
-			Labels:      internal.ControlPlaneMachineLabelsForCluster(kcp, cluster.Name),
-			Annotations: map[string]string{},
-			// Note: by setting the ownerRef on creation we signal to the Machine controller that this is not a stand-alone Machine.
+			UID:       machineUID,
+			Name:      machineName,
+			Namespace: kcp.Namespace,
 			OwnerReferences: []metav1.OwnerReference{
 				*metav1.NewControllerRef(kcp, controlplanev1.GroupVersion.WithKind("KubeadmControlPlane")),
 			},
+			Labels:      map[string]string{},
+			Annotations: map[string]string{},
 		},
 		Spec: clusterv1.MachineSpec{
 			ClusterName:       cluster.Name,
-			Version:           &kcp.Spec.Version,
+			Version:           version,
+			FailureDomain:     failureDomain,
 			InfrastructureRef: *infraRef,
 			Bootstrap: clusterv1.Bootstrap{
 				ConfigRef: bootstrapRef,
 			},
-			FailureDomain:           failureDomain,
-			NodeDrainTimeout:        kcp.Spec.MachineTemplate.NodeDrainTimeout,
-			NodeDeletionTimeout:     kcp.Spec.MachineTemplate.NodeDeletionTimeout,
-			NodeVolumeDetachTimeout: kcp.Spec.MachineTemplate.NodeVolumeDetachTimeout,
 		},
 	}
 
-	// In case this machine is being created as a consequence of a remediation, then add an annotation
-	// tracking remediating data.
-	// NOTE: This is required in order to track remediation retries.
-	if remediationData, ok := kcp.Annotations[controlplanev1.RemediationInProgressAnnotation]; ok {
-		machine.Annotations[controlplanev1.RemediationForAnnotation] = remediationData
-	}
+	// Set the in-place mutable fields.
+	// When we create a new Machine we will just create the Machine with those fields.
+	// When we update an existing Machine will we update the fields on the existing Machine (in-place mutate).
 
-	// Machine's bootstrap config may be missing ClusterConfiguration if it is not the first machine in the control plane.
-	// We store ClusterConfiguration as annotation here to detect any changes in KCP ClusterConfiguration and rollout the machine if any.
-	clusterConfig, err := json.Marshal(kcp.Spec.KubeadmConfigSpec.ClusterConfiguration)
-	if err != nil {
-		return errors.Wrap(err, "failed to marshal cluster configuration")
-	}
+	// Set labels
+	desiredMachine.Labels = internal.ControlPlaneMachineLabelsForCluster(kcp, cluster.Name)
 
+	// Set annotations
 	// Add the annotations from the MachineTemplate.
 	// Note: we intentionally don't use the map directly to ensure we don't modify the map in KCP.
 	for k, v := range kcp.Spec.MachineTemplate.ObjectMeta.Annotations {
-		machine.Annotations[k] = v
-	}
-	machine.Annotations[controlplanev1.KubeadmClusterConfigurationAnnotation] = string(clusterConfig)
-
-	if err := r.Client.Create(ctx, machine); err != nil {
-		return errors.Wrap(err, "failed to create machine")
+		annotations[k] = v
```

Comment:
Can we write those annotations here directly on the Machine and add the ones from `annotations` after that?

This ensures that "our" annotations cannot be overwritten by the ones from the MachineTemplate